### ISSUE TITLE
Could provoke a PHP Fatal error ([] operator not supported for strings) triggered when it should not

### ISSUE THREAD

**Issue by Mandy91** [kalessil/phpinspectionsea#1178]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                       |
| Plugin         | Php Inspections (EA Extended) v3.0.10 |
| Language level | PHP 7.2 |

In some cases _Could provoke a PHP Fatal error ([] operator not supported for strings)_ is triggered when it should not.

![image](https://user-images.githubusercontent.com/2277717/51521035-a4dbc400-1e25-11e9-864d-34dd54df1011.png)

```
class A {

	protected $prefix = 'abc';
	protected $array = [];
	
	public function __construct(): void {
		$_POST[$this->prefix][] = $this->prefix;

		$this->array[$this->prefix][] = $this->prefix;
	}
	
}


$prefix = 'abc';
$_POST[$prefix][] = $prefix;

$prefix = 'abc';
$array = [];
$array[$prefix][] = $prefix;
```

No error should be triggered

PhpStorm 2018.3.3 (Installed from Jetbrains Toolbox)
Fedora 28, KDE Plasma

**Comment by kalessil** [comment]
Thank you for reporting @Mandy91 , I'll take a look why this gets reported. Agree that superglobals should not be reported in this context.

**Comment by kalessil** [comment]
Fixed!

**Comment by shehi** [comment]
@kalessil , this problem is back in PHPStorm 2020.

**Comment by watain666** [comment]
> 
> 
> @kalessil , this problem is back in PHPStorm 2020. EA Extended 4.0.4.1.
> 
> ![image](https://user-images.githubusercontent.com/653569/85952073-39c1e580-b967-11ea-8deb-02b5feae837a.png)

Same herre
![image](https://user-images.githubusercontent.com/19923168/87637785-563a8c00-c775-11ea-9a1d-74949f81d59a.png)

PHPStorm 2020.1.2
EA Extended 4.0.5

**Comment by guipassos** [comment]
The is is back in version 2020.3.2
